qforce/qm/qm.py

```diff
@@ -61,20 +67,151 @@ def read_scan(self, files):
+    def _xtb_torsiondrive_read(log_file):
+        '''Read the TorsionDrive output.
+
+        Parameters
+        ----------
+        config : config
+            A configparser object with all the parameters.
+
+        Returns
+        -------
+        n_atoms : int
+            The number of atoms in the molecule.
+        coords : list
+            A list of array of float. The list has the length of the number
+            of steps and the array has the shape of (n_atoms, 3).
+        angles : list
+            A list (length: steps) of the angles that is being scanned.
+        energies : list
+            A list (length: steps) of the energy.
+        point_charges : dict
+            A dictionary with key in charge_method and the value to be a
+            list of float of the size of n_atoms.
+        '''
+        frames = read(log_file, index=':', format='extxyz')
+        n_atoms = len(frames[0])
+        energy_list = []
+        coord_list = []
+        angle_list = []
+        for frame in frames:
+            coord_list.append(frame.positions)
+            _, angle, _, energy = list(frame.info.keys())
+            angle = float(angle[1:-2])
+            angle_list.append(angle)
+            energy = float(energy)
+            energy_list.append(energy)
+
+        point_charges = np.loadtxt(os.path.splitext(log_file)[0]+'.charges')
+        energies = np.array(energy_list) * Hartree * mol / kJ
+        return n_atoms, coord_list, angle_list, energies, {'xtb': point_charges}
+
+    def _xtb_torsiondrive_write(self, file, dir, scan_id, coords, atnums,
+                                   scanned_atoms, charge,
+                                   multiplicity):
+        '''Create the TorsionDrive input.
+
+        Parameters
+        ----------
+        file : file
+            The file object to write the torsiondrive command
+        dir : string
+            The path to the current directory.
+        scan_id : string
+            The name of the fragment.
+        coords : numpy.array
+            Array of the coordinates in the same of (n,3), where n is the
+            number of atoms.
+        atnums : list
+            A list of n integers representing the atomic number, where n is
+            the number of atoms.
+        scanned_atoms : list
+            A list of 4 integers representing the one-based atom index of the
+            dihedral.
+        charge : int
+            The total charge of the fragment.
+        multiplicity : int
+            The multiplicity of the molecule.
+         '''
+        try:
+            os.makedirs(f'{dir}/{scan_id}_torsiondrive')
+        except FileExistsError:
+            warn(f'Folder {dir}/{scan_id}_torsiondrive exists, remove the folder.')
+            shutil.rmtree(f'{dir}/{scan_id}_torsiondrive')
+            os.makedirs(f'{dir}/{scan_id}_torsiondrive')
+
+        # Create the coordinate input file
+        with open(f'{dir}/{scan_id}_torsiondrive/input.xyz', 'w') as f:
```

Comment:
perhaps you can use ASE here again to write the XYZ file?

See an example [here](https://github.com/selimsami/qforce/blob/06d13322f0a996a9996f2c8ff4613d703420ea53/qforce/qm/qm.py#L136).

Reply:
@selimsami Thanks. I was thinking of the same thing but I couldn't get the mol.info to write what I want. It is nice that I could just put it in the comment.

Reply:
@selimsami Sorry about more ase question. I wonder how do I create an ase mol object given ele_id and pos? see the input here https://github.com/selimsami/qforce/pull/43/files#diff-f579a382b4ca6ded95ab6837635635e6329d9fe38329927c5f4e2d8a4ff2b243R119

Reply:
from ase import Atoms

`mol = Atoms(positions=pos, numbers=ele_id)`